src/app/api/collections/tree.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+from uuid import UUID
+
+from aiohttp import ClientSession
+from fastapi import Path
+from pydantic import BaseModel
+
+from app.core.constants import PORTAL_ROOT_ID, PORTALS
+
+
+def portal_id_with_root_param(
+    *,
+    node_id: UUID = Path(
+        ...,
+        examples={
+            "Alle Fachportale": {"value": PORTAL_ROOT_ID},
+            **PORTALS,
+        },
+    ),
+) -> UUID:
+    return node_id
+
+
+class PortalTreeNode(BaseModel):
+    noderef_id: UUID
+    title: str
+    children: list[PortalTreeNode]
+
```

Comment:
Good suggestion - since this code is a byproduct of the current PR and not the main focus, I added it to https://github.com/openeduhub/metaqs-main/issues/44